File: CSharp/MongoDB/REST/Client/REST_Response.cs
namespace MongoDB.REST.Client;

public class REST_Response
{
    public int StatusCode { get; }
    public string ContentType { get; }
    public string Body { get; }

    public REST_Response(int statusCode, string contentType, string body)
    {
        StatusCode = statusCode;
        ContentType = contentType;
        Body = body;
    }

    public override string ToString()
    {
        return $"RestResponse(status_code={StatusCode}, content_type='{ContentType}', body='{Body}')";
    }

    public Dictionary<string, object> ToDict()
    {
        return new Dictionary<string, object>
        {
            { "status_code", StatusCode },
            { "content_type", ContentType },
            { "body", Body }
        };
    }
}

File: Examples/Rust/Containers/MongoDB-REST/src/src/services/mongodb/rest/rest_response.rs
pub struct RESTResponse {
    pub status_code: u16,
    pub content_type: String,
    pub body: String,
}

impl RESTResponse {
    pub fn new(status_code: u16, content_type: &str, body: String) -> Self {
        RESTResponse {
            status_code,
            content_type: content_type.to_string(),
            body,
        }
    }
}